### ISSUE TITLE
Docs for `dt-struct/map->struct!` and `dt-struct/map->struct`

### ISSUE THREAD

**Issue by phronmophobic** [cnuernber/dtype-next#110]
These [struct helpers](https://github.com/cnuernber/dtype-next/blob/c0c25273e78eeb79564883a62aaf9d42493ed665/src/tech/v3/datatype/struct.clj#L454) seem really useful.

Also, I'm not sure if makes sense, but I think a 2 arity version of `map->struct` would be helpful:
```clojure
(defn map->struct
  ([dtype data]
   (map->struct dtype data :gc))
  ([dtype data track-type]
   (map->struct! data (new-struct dtype {:container-type :native-heap
                                         :resource-type track-type}))))
```

If not, then I would suggest that the doc string also document the relevant values for `:resource-type`. Otherwise, you have to follow read the doc strings for dt-struct/new-struct, native-buffer/malloc, and `tech.v3.resource/track`.

**Comment by phronmophobic** [comment]
See https://github.com/cnuernber/dtype-next/pull/111

**Comment by cnuernber** [comment]
This was addressed by #111
